src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -305,37 +322,41 @@ private static string GetFullyQualifiedTypeName(TypeDefinition typeDef, PEModule
             }
         }
 
-        private sealed class Extensions<TExtension> where TExtension : class
+        private sealed class Extensions<TExtension, TAttribute>
+            where TExtension : class
+            where TAttribute : Attribute
```

Comment:
>TAttribute [](start = 18, length = 10)

Is this type parameter used outside of the constructor? If not, consider passing the namespace and name explicitly to the constructor, or create a generic `static Extensions<TExtension> Create<TExtension, TAttribute>(...)` factory method.

Reply:
I refactored it just pass in the type as a param instead.